File: src/main/java/uss/code/registration/dto/response/RegistrationResponse.java
package uss.code.registration.dto.response;

import lombok.Builder;
import uss.code.course.domain.Course;
import uss.code.course.dto.response.CourseResponse;

import static lombok.AccessLevel.PRIVATE;

@Builder(access = PRIVATE)
public record RegistrationResponse(
        CourseResponse courseResponse
) {
    public static RegistrationResponse from(final Course course) {
        return RegistrationResponse.builder()
                .courseResponse(CourseResponse.from(course))
                .build();
    }
}

File: src/main/kotlin/uss/code/registration/dto/response/RegistrationResponse.kt
package uss.code.registration.dto.response

import uss.code.course.domain.Course
import uss.code.course.dto.response.CourseResponse

data class RegistrationResponse(
    val courseResponse: CourseResponse,
) {
    companion object {
        fun from(course: Course): RegistrationResponse {
            return RegistrationResponse(
                courseResponse = CourseResponse.from(course),
            )
        }
    }
}
